account_analytic_compassion/models/account_asset.py

```diff
@@ -0,0 +1,53 @@
+##############################################################################
+#
+#    Copyright (C) 2020 Compassion CH (http://www.compassion.ch)
+#    @author: William Martin <wmartin@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import models, fields
+
+
+class AccountAsset(models.Model):
+    _inherit = 'account.asset'
+
+    # Many2many field to account.analytic.tag table
+    analytic_tag_ids = fields.Many2many(
+        comodel_name='account.analytic.tag',
+        string='Analytic Tag',
+    )
+
+
+class AccountAssetLine(models.Model):
+    _inherit = 'account.asset.line'
+
+    # override of parent method _setup_move_line_data
+    # inherited from object AccountAssetLine
+    # at account-financial-tools/account_asset_management/models/account_asset_line.py
+    def _setup_move_line_data(self, depreciation_date, account, ml_type, move):
+        asset = self.asset_id
+        amount = self.amount
+        analytic_id = False
+        if ml_type == 'depreciation':
+            debit = amount < 0 and -amount or 0.0
+            credit = amount > 0 and amount or 0.0
+        elif ml_type == 'expense':
+            debit = amount > 0 and amount or 0.0
+            credit = amount < 0 and -amount or 0.0
+            analytic_id = asset.account_analytic_id.id
+        move_line_data = {
+            'name': asset.name,
+            'ref': self.name,
+            'move_id': move.id,
+            'account_id': account.id,
+            'credit': credit,
+            'debit': debit,
+            'journal_id': asset.profile_id.journal_id.id,
+            'partner_id': asset.partner_id.id,
+            'analytic_account_id': analytic_id,
+            'date': depreciation_date,
+            'asset_id': asset.id,
+            'analytic_tag_ids': asset.analytic_tag_ids.ids
+        }
+        return move_line_data
```

Comment:
It is not a good practice to completely rewrite a method, instead of making usage of the super call. The parent method returns a dictionary. You can use super to have the result, modify the result in your method, and then return the final result, like this:
```suggestion
        move_line_data = super()._setup_move_line_data(
            depreciation_date, account, ml_type, move)
        if ml_type == 'expense':
            move_line_data["analytic_tag_ids"] = [(6, 0, asset.analytic_tag_ids.ids)]
        return move_line_data
```
[(6, 0, asset.analytic_tag_ids.ids)] is a special command for writing in many2many field: see https://www.odoo.com/documentation/12.0/reference/orm.html#odoo.models.Model.write for reference

Reply:
Tried:
```python
 move_line_data["analytic_tag_ids"] = [(6, 0, asset.analytic_tag_ids.ids)]
```
And Travis tests did not pass.
changed to:
```python
move_line_data["analytic_tag_ids"] = asset.analytic_tag_ids.ids
```